extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeCommentsExtractor.java

```diff
@@ -224,24 +242,23 @@ private void collectCommentsFrom(final CommentsInfoItemsCollector collector,
             throw new ParsingException("Unable to get parse youtube comments", e);
         }
 
-        for (final Object c : comments) {
-            if (c instanceof JsonObject) {
-                final CommentsInfoItemExtractor extractor = new YoutubeCommentsInfoItemExtractor(
-                        (JsonObject) c, getUrl(), getTimeAgoParser());
-                collector.commit(extractor);
-            }
-        }
+        final String url = getUrl();
+        comments.stream()
+                .filter(JsonObject.class::isInstance)
+                .map(JsonObject.class::cast)
+                .map(jObj -> new YoutubeCommentsInfoItemExtractor(jObj, url, getTimeAgoParser()))
```

Comment:
```suggestion
        comments.stream()
                .filter(JsonObject.class::isInstance)
                .map(JsonObject.class::cast)
                .map(jObj -> new YoutubeCommentsInfoItemExtractor(jObj, getUrl(), getTimeAgoParser()))
```

Reply:
That's not possible since it throws an exception which can't be handled in the lambda (because it's not a RuntimeException).
![grafik](https://user-images.githubusercontent.com/40789489/155012399-1fcf67c4-ddb1-4852-9eec-c4d6582c6273.png)